packages/client/src/mud/setup.ts

```diff
@@ -14,7 +14,8 @@ export async function setup() {
     networkConfig,
     world,
     contractComponents,
-    {} as never
+    {} as never,
+    { syncThread: "main" }
```

Comment:
isn't this the default behavior?

Reply:
no, because the previous default was syncing in a worker and I didn't want a "breaking change" (maybe syncing in the main thread breaks things for existing apps with a lot of events). We can make it the default in a follow up (either in v2 or just when we had time to test in some high throughput apps).